RoboJackets/robocup-software
# RobotFactoryPosition: Infrastructure and Restarts

## Description
Updates positions based on game state (e.g. Penalty, Kickoff, Free Kick)

## Associated / Resolved Issue
Resolves # or [ClickUp card](https://app.clickup.com/t/86azbvek6)
Resolves # or [ClickUp card](https://app.clickup.com/t/86aznadc5)

## Steps to Test
Use external ref to call for kickoffs, penalties, and free kicks for both teams and verify rule-compliance.

## Key Files to Review
 * robot_factory_position.cpp
 * robot_factory_position.hpp
 * offense.cpp
 * offense.hpp
 * penalty_non_kicker.cpp
 * penalty_non_kicker.hpp
 * penalty_player.cpp
 * penalty_player.hpp
 * smartidling.cpp
 * smartidling.hpp
 * free_kicker.cpp
 * free_kicker.hpp

## Review Checklist


## (Optional) Sub-issues (for drafts)
_Note: if you find yourself breaking this PR into many smaller features, it may make sense to break up the PR into logical units based on these features._
